src/Job/StaticCacheFullBuildJob.php

```diff
@@ -80,24 +83,26 @@ public function process(): void
         }
 
         if (count($this->URLsToProcess) === 0) {
-            $trimSlashes = function ($value) {
-                $value = trim($value, '/');
+            $cleanUrl = function ($value) {
 
                 // We want to trim the schema from the beginning as they map to the same place
                 // anyway.
                 $value = ltrim($value, 'http://');
                 $value = ltrim($value, 'https://');
+                $value = preg_replace('#[\?&]stage=' . Versioned::LIVE . '#', '', $value);
```

Comment:
This was a necessary bug fix for me to run locally without it trigger an infinite loop.  It's possibly related to running framework 4.12, though I'm not entirely sure.